packages/storage-plus/src/snapshot/item.rs

```diff
@@ -45,6 +46,11 @@ impl<'a, T> SnapshotItem<'a, T> {
     pub fn remove_checkpoint(&self, store: &mut dyn Storage, height: u64) -> StdResult<()> {
         self.snapshots.remove_checkpoint(store, height)
     }
+
+    pub fn changelog(&self) -> Map<u64, ChangeSet<T>> {
+        // Build and return a compatible Map with the proper key type
+        Map::new(unsafe { from_utf8_unchecked(self.snapshots.changelog.namespace()) })
```

Comment:
Why not expose this functionality as a method on Snapshot? (`self.snapshots.changelog()`)

Reply:
No, because I'm overriding the `Map` key type here. Our empty (`Unit`) keys have a particular behaviour, in that they are not length-prefixed zero-length elements in tuples (they are just being skipped, when building the key).
 
What I can do to improve on this is to save the changelog namespace in this class, so that I can use it here directly, and avoid the conversion to unicode. Will do.